From f8e8f1daf12e30d3197b25d6bd10c03d1f932e20 Mon Sep 17 00:00:00 2001
From: Nick Clifton <nickc@redhat.com>
Date: Tue, 27 Sep 2016 12:08:19 +0100
Subject: [PATCH] Ensure that the timestamp in PE/COFF headers is always
 initialised.

	PR ld/20634
	* peXXigen.c (_bfd_XXi_only_swap_filehdr_out): Put 0 in the
	timestamp field if real time values are not being stored.
---
 bfd/peXXigen.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/bfd/peXXigen.c b/bfd/peXXigen.c
index c92c1ea..d713d96 100644
--- a/bfd/peXXigen.c
+++ b/bfd/peXXigen.c
@@ -879,6 +879,8 @@ _bfd_XXi_only_swap_filehdr_out (bfd * abfd, void * in, void * out)
   /* Only use a real timestamp if the option was chosen.  */
   if ((pe_data (abfd)->insert_timestamp))
     H_PUT_32 (abfd, time (0), filehdr_out->f_timdat);
+  else
+    H_PUT_32 (abfd, 0, filehdr_out->f_timdat);
 
   PUT_FILEHDR_SYMPTR (abfd, filehdr_in->f_symptr,
 		      filehdr_out->f_symptr);
-- 
2.1.4

